src/collective/ifttt/browser/views.py

```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+from plone import api
+from Products.Five.browser import BrowserView
+
+
+class checkView(BrowserView):
+    def check_iftttconfig(self):
+
+        # check if IFTTT secret key is added
+        secret_key = api.portal.get_registry_record('ifttt.ifttt_secret_key')
+        if not secret_key:
+            return False
+        return True
```

Comment:
I believe that you should also check for all conditions checked here https://github.com/plone/plone.app.contentrules/blob/master/plone/app/contentrules/browser/info.py#L11 the check that content rules can be used in general.

Technically you could look up that view to do the checks, but the code could be easier to maintain if you just copy all that code into your view here.

Oh, remember to exclude these line
```
        if not storage:
            return False
```

because that is only True when the folder already has rules assigned.

Reply:
@datakurre  I believe this permission is already been checked here?
https://github.com/collective/collective.ifttt/pull/47/files#diff-53d7c6f572e6ce73da2975050926eb3eR12

Reply:
@Shriyanshagro That's correct. The permission is already checked. But there are still a couple of other checks in that plone.app.contentrules info view:

```python
        if not IRuleAssignable.providedBy(self.context):
            return False
```

check that we are on a folder where content rules can be assigned.

```
        storage = queryUtility(IRuleStorage)
        if not storage.active:
            return False
```

checks that content rules are not globally disabled from content rules control panel.

Because there are that many conditions, it would not harm all to to repeat those all in that helper view to make one place for all the conditions (and maybe even remove the permissions checks from actions.xml lines to prevent duplicate checks).